libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2398,6 +2398,38 @@ void TypeChecker::typeCheckBytesConcatFunction(
 	}
 }
 
+void TypeChecker::typeCheckERC7201Builtin(FunctionCall const& _functionCall, FunctionType const* _functionType)
+{
+	if (_functionCall.arguments().size() > 0)
+	{
+		auto const* literalArg = dynamic_cast<Literal const*>(_functionCall.arguments()[0].get());
+		auto const* variableString = dynamic_cast<Identifier const*>(_functionCall.arguments()[0].get());
+		bool validArg = false;
+		std::string errorMsg;
+
+		if (
+			(
+				literalArg &&
+				dynamic_cast<StringLiteralType const*>(literalArg->annotation().type)
+			) ||
+			(
+				variableString &&
+				dynamic_cast<ArrayType const*>(variableString->annotation().type) &&
+				dynamic_cast<ArrayType const*>(variableString->annotation().type)->isString() &&
+				*variableString->annotation().isConstant
+			)
+		)
```

Comment:
I don't think we need to limit it to `Literal` and `Identifier`. We're fine with anything that has the right type and is not mutable.

```suggestion
		if (
			dynamic_cast<StringLiteralType const*>(argument->annotation().type) ||
			(
				*argument->annotation().isConstant &&
				dynamic_cast<ArrayType const*>(argument->annotation().type) &&
				dynamic_cast<ArrayType const*>(argument->annotation().type)->isString()
			)
		)
```

Note that your version excludes e.g. `MemberAccess`. I guess it might still not evaluate properly due to the dependency issue we have but it should at least pass through here without problems.